pulp_2_tests/tests/rpm/api_v2/test_modularity.py

```diff
@@ -200,3 +204,68 @@ def test_all(self):
             with self.subTest(package=key):
                 module = [line for line in lines if key in line]
                 self.assertEqual(len(module), value, module)
+
+
+class UploadModuleTestCase(unittest.TestCase):
+    """Upload a module.yaml file and test upload import in Pulp repo."""
+
+    def test_all(self):
+        """Verify whether uploaded module.yaml is reflected in the pulp repo."""
+        cfg = config.get_config()
+        if cfg.pulp_version < Version('2.17'):
+            raise unittest.SkipTest('This test requires at least Pulp 2.17 or newer.')
+        if not os_support_modularity(cfg):
+            raise unittest.SkipTest('This test requires an OS that supports modularity.')
+        client = api.Client(cfg, api.json_handler)
+        # Create a normal Repo without any data.
+        body = gen_repo(
+            importer_config={'feed': RPM_UNSIGNED_FEED_URL},
+            distributors=[gen_distributor()]
+        )
+        repo = client.post(REPOSITORY_PATH, body)
+        repo = client.get(repo['_href'], params={'details': True})
+        self.addCleanup(client.delete, repo['_href'])
+        sync_repo(cfg, repo)
+
+        # download modules.yaml and upload it to pulp_repo
+        unit = self._get_module_yaml_file(cfg, RPM_WITH_MODULES_FEED_URL)
+        upload_import_unit(cfg, unit, {
+            'unit_key': {},
+            'unit_type_id': 'modulemd',
+        }, repo)
+        repo = client.get(repo['_href'], params={'details': True})
+        # Assert that `modulemd` and `modulemd_defaults` are present on the
+        # repository.
+        self.assertIsNotNone(repo['content_unit_counts']['modulemd'])
+        self.assertIsNotNone(repo['content_unit_counts']['modulemd_defaults'])
+
+    @staticmethod
+    def _get_module_yaml_file(cfg, path):
+        """Return the path to ``modules.yaml``, relative to repository root.
+
+        Given a detailed dict of information about a published, repository,
+        parse that repository's ``repomd.xml`` file and tell the path to the
+        repository's ``[…]-modules.yaml`` file. The path is likely to be in the
+        form ``repodata/[…]-modules.yaml.gz``.
+        """
+        client = api.Client(cfg, xml_handler)
+        repo_path = urljoin(path, 'repodata/repomd.xml')
+        root_elem = client.get(repo_path)
```

Comment:
API client was designed to ease interactions with Pulp. Using the API client to talk with another resource will generate warnings. 
``` bash
/lib64/python3.6/site-packages/pulp_smash/api.py:477: RuntimeWarning: This client should be used to communicate with c7.pulp.vm, but a request is being made to repos.fedorapeople.org. The request will be made, but beware that information intended for c7.pulp.vm (such as authentication tokens) may now be sent to repos.fedorapeople.org. Here's the full list of options being sent with this request: {'verify': False, 'auth': ('admin', 'admin'), 'url': 'https://repos.fedorapeople.org/pulp/pulp/fixtures/rpm-with-modules/repodata/repomd.xml'}
  RuntimeWarning
```
Perhaps this API client call be can be replace by something like
```python
        response = utils.http_get(urljoin(path, 'repodata/repomd.xml'))
        root_elem = ElementTree.fromstring(response)
```